src/main/features/claimant-response/routes/payment-date.ts

```diff
@@ -48,30 +59,59 @@ class PaymentDatePage extends AbstractPaymentDatePage<DraftClaimantResponse> {
 
   generateCourtCalculatedPaymentIntention (draft: Draft<DraftClaimantResponse>, claim: Claim, decisionType: DecisionType): PaymentIntention {
     const courtCalculatedPaymentIntention = new PaymentIntention()
+    const claimResponse = claim.response as FullAdmissionResponse | PartialAdmissionResponse
+    draft.document.courtDecisionType = decisionType
 
     if (decisionType === DecisionType.CLAIMANT) {
-      courtCalculatedPaymentIntention.paymentDate = draft.document.alternatePaymentMethod.paymentDate.date.toMoment()
-      courtCalculatedPaymentIntention.paymentOption = draft.document.alternatePaymentMethod.toDomainInstance().paymentOption
-
+      if (draft.document.alternatePaymentMethod.paymentOption.option.value === PaymentOption.BY_SPECIFIED_DATE) {
+        courtCalculatedPaymentIntention.paymentDate = draft.document.alternatePaymentMethod.toDomainInstance().paymentDate
+        courtCalculatedPaymentIntention.paymentOption = PaymentOption.BY_SPECIFIED_DATE
+      }
       return courtCalculatedPaymentIntention
     }
 
-    if (decisionType === DecisionType.COURT || decisionType === DecisionType.DEFENDANT) {
+    if (decisionType === DecisionType.COURT) {
       const paymentPlanFromDefendantFinancialStatement: PaymentPlan = PaymentPlanHelper.createPaymentPlanFromDefendantFinancialStatement(claim)
       const lastPaymentDate: Moment = paymentPlanFromDefendantFinancialStatement.calculateLastPaymentDate()
 
-      if (draft.document.alternatePaymentMethod.paymentOption.option.value === PaymentOption.BY_SPECIFIED_DATE) {
+      if (claimResponse.paymentIntention.paymentOption === PaymentOption.BY_SPECIFIED_DATE) {
         courtCalculatedPaymentIntention.paymentDate = lastPaymentDate
-        courtCalculatedPaymentIntention.paymentOption = draft.document.alternatePaymentMethod.toDomainInstance().paymentOption
+        courtCalculatedPaymentIntention.paymentOption = PaymentOption.BY_SPECIFIED_DATE
       }
 
```

Comment:
suggestion: extract things in if conditions to a method so they are more readable

Reply:
yes, completely agree. We are planning to calculate completion date and length of payment as part of payment plan creation then lot of this code will be simplified. Until payment plan has everything bundled up in 1, this code will be verbose.